provider.go

```diff
@@ -1052,55 +1072,346 @@ func (p *provider) Delete(ctx context.Context, req *rpc.DeleteRequest) (*emptypb
 
 }
 
+// ConstructRequest captures enough data to be able to register nested components against the caller's resource
+// monitor.
 type ConstructRequest struct {
-	URN       presource.URN
-	Preview   bool
-	Construct func(context.Context, ConstructFunc) (ConstructResponse, error)
+	Urn     presource.URN // the Pulumi URN for this resource.
+	Info    ConstructInfo
+	Parent  presource.URN
```

Comment:
I don't understand the logic where `Parent` is top-level instead of `Info` or `Options`, and I don't really understand why we have both `Info` and `Options`. Can you explain?

Reply:
Yes, the structure was made to resemble [plugin.ConstructRequest](https://github.com/pulumi/pulumi/blob/b5329063ffb665a49c534a4994b4eddbd6acef5f/sdk/go/common/resource/plugin/provider.go#L290-L297), which seemed to be a high-level representation of the information. I have a mild preference for flattening it (dropping Info and Options), perhaps I should simply do that.

Reply:
I've flattened it, thanks.